content/en/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md

```diff
@@ -396,13 +396,10 @@ have a number of fields that you can use to more precisely control the behavior
 liveness and readiness checks:
 
 * `initialDelaySeconds`: Number of seconds after the container has started before startup,
-  liveness or readiness probes are initiated. If a startup probe is defined, liveness and
-  readiness probe delays do not begin until the startup probe has succeeded.
-  An `initialDelaySeconds` value greater than 0 will delay when the first probe starts. If
-  the value of `periodSeconds` is greater than `initialDelaySeconds` then `periodSeconds`
-  intervals remain effective. For example, if `initialDelaySeconds` is set to 10, but
-  `periodSeconds` is set to 20, periodic probes will still begin after an initial 20 seconds.
-  Defaults to 0 seconds. Minimum value is 0.
+  liveness or readiness probes are effective. If a startup probe is defined, liveness and
+  readiness probe delays do not begin until the startup probe has succeeded. If the value of
+  `periodSeconds` is greater than `initialDelaySeconds` then the `initialDelaySeconds` will
+  be included in the first period interval. Defaults to 0 seconds. Minimum value is 0.
```

Comment:
If the value of `periodSeconds` is greater than `initialDelaySeconds` then the `initialDelaySeconds` will be ignored. The delayed probes will start after the end
of the first `periodSeconds`. However, when `initialDelaySeconds` is omitted
(defaulted to 0), the liveness, readiness probes are not delayed and they will start
immediately after the container has started.


Is this what you were trying to clarify?

Reply:
> If the value of periodSeconds is greater than initialDelaySeconds then the initialDelaySeconds will be ignored. 

I think we are on the same page in terms of the actual behavior here but I would not say it will be "ignored". My understanding is that if `0 < initialDelaySeconds < periodSeconds`, then the "initial" probe immediately after container start will be skipped due to `initialDelaySeconds`. It is not "ignored"; it has a clear effect compared to the field not being set / default to 0. I think what the author was trying to express was, as long as `0 < initialDelaySeconds < periodSeconds`, it does not matter what exact value you specify for `initialDelaySeconds`, the first probe will always happen at the end of first `periodSeconds`. However it is quite different from "it is ignored"

Reply:
Yes, I intend to change the wording of _"are initiated"_ and _"will be ignored"_, because they are misleading and do not match the actual behavior, especially for the case `initialDelaySeconds=0`.
The issue is: based on the Doc, I tried to remove `initialDelaySeconds` for value less than `periodSeconds`, because Doc says it will be ignored anyway. Then the probe happened immediately, not after `periodSeconds`, which is not expected from the Doc.

Reply:
@qiyueyao I take it as, for people writing pod specs and wonder if they can/should set `initialDelaySeconds` to be a value less than `periodSeconds`, we don't want them to read the doc as is currently and think they can just remove the `initialDelaySeconds` which will be ignored anyways. As a fact, it will cause the "0th" probe to delay for `periodSeconds`

Reply:
@qiyueyao, I tested your observation.

I configured my liveness probe to the below.
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: probe-test
  labels:
    app: probe-test
spec:
  replicas: 1
  selector:
    matchLabels:
      app: probe-test
  template:
    metadata:
      labels:
        app: probe-test
    spec:
      containers:
      - name: nginx
        image: nginx:latest
        ports:
        - containerPort: 80
        livenessProbe:
          httpGet:
            path: /
            port: 80
          initialDelaySeconds: 0
          periodSeconds: 5
          timeoutSeconds: 1   
```
The `initialDelaySeconds` value was ignored while `periodSeconds` took effect. Observe the time difference between when the container started and when it received the liveness probe, **09:42:43** and **09:42:47**, a 4s difference. I noticed that 1s is always subtracted from the `periodSeconds` value when `initialDelaySeconds` has a value of 0, I don't know why.

```bash
2024/12/08 09:42:43 [notice] 1#1: start worker process 28
172.30.2.2 - - [08/Dec/2024:09:42:47 +0000] "GET / HTTP/1.1" 200 615 "-" "kube-probe/1.31" "-"
```

When I used the below.
```yaml
          initialDelaySeconds: 4
          periodSeconds: 5
          timeoutSeconds: 1   
```

I got this.
```bash
2024/12/08 09:51:20 [notice] 1#1: start worker process 28
172.30.2.2 - - [08/Dec/2024:09:51:25 +0000] "GET / HTTP/1.1" 200 615 "-" "kube-probe/1.31" "-"
```
The `initialDelaySeconds` was also ignored.

Your observation may be wrong. Could you share a sample config of what you tried?